### DISCUSSION TITLE
site_description translation (i18n)

### DISCUSSION THREAD

**Discussion by why-not-try-calmer** [squidfunk/mkdocs-material#5359] [Q&A] [answered]
I wondered whether there was an easy way to add translation to the `mkdocs.yml:site_description` attribute / key. Is this supported out of the box? If not is there as known workaround?

Thanks a lot for this very good plugin!

**Comment by squidfunk** [comment]
Are you referring to [mkdocs-static-i18n](https://github.com/ultrabug/mkdocs-static-i18n)? If so, better ask the maintainers of the plugin, we don't maintain it here. We only maintain the theme Material for MkDocs 😉 Other than that, here's some guidance how to build a multi-language site:

- https://github.com/squidfunk/mkdocs-material/discussions/2346
- https://github.com/ultrabug/mkdocs-static-i18n/issues/191

**Comment by kamilkrzyskow** [comment, marked as answer]
Using the recommended way to implement i18n into your docs, with separate mkdocs.yml files would easily allow you to use separate `site_description` for each of the language versions.

If you're using the i18n plugin for that then it's currently not available. I'll be adding it this month if the plugin's author won't finish [the refactor](https://github.com/ultrabug/mkdocs-static-i18n/pull/216) sooner.

**Reply by squidfunk** [reply]
Nice! Looking forward to see how you solve this within the plugin, as this seems to be one of the biggest ones. Still not 100% sold on any of the multi-language approaches, which is why we have nothing in our docs that we feel good recommending.

**Reply by why-not-try-calmer** [reply]
Thanks, that's what I was worried about. But @kamilkrzyskow's answer makes me very optimistic! Thanks a lot!

**Reply by kamilkrzyskow** [reply]
Hey, @why-not-try-calmer 
the refactor will take some more time, so I added it myself. The changes are on my fork. Tested only with the `suffix` structure mode of the plugin, since that is what I'm using.

GMC uses a `requirements.txt` file to manage `pip` dependencies.
To use my fork instead of the main package we select a specific hash commit, you can also comment out the previous package with `#`
```txt
# mkdocs-static-i18n==0.56
git+https://github.com/kamilkrzyskow/i18n.git@<COMMIT_HASH>#egg=mkdocs_static_i18n
```

Here are some branches from my fork with increasing amount of changes, the last one is a cumulative branch for all changes applicable to our Gothic Modding Community documentation.
https://github.com/kamilkrzyskow/i18n/commits/site-description
https://github.com/kamilkrzyskow/i18n/commits/social-plugin-support
https://github.com/kamilkrzyskow/i18n/commits/i18n-gmc

So we're using `git+https://github.com/kamilkrzyskow/i18n.git@5e286be68ad6fafed7affd44cd57d3625c0a02cc#egg=mkdocs_static_i18n`
in the `requirements.txt`

**Reply by why-not-try-calmer** [reply]
Thanks very much for the nice work and for keeping me in the loop; will test it and let you know if I have valuable feedback!

**Reply by why-not-try-calmer** [reply]
Hi again  @kamilkrzyskow, wholehearted thank YOU for providing the `site_description` config parameter. Just pinned several repositories to it. also I hope the large PR for revamping the plugin goes well!

**Reply by kamilkrzyskow** [reply]
Hi @why-not-try-calmer,
I'm currently on holiday and rather out of the loop. I hope there are no issues on your end, since I won't be able to fix them. Afaik no one reported any issues to me concerning the Gothic Modding Community documentation over the last few months.

> Please note that my changes aren't compatible with the Social cards plugin from the Insider version of MkDocs Material.

The PR is on hold for now, because ultrabug works on MkDocs 1.5.0 and wants to make it more future proof first before modifying the plugin. 

I'm sure everything will sort itself out after some time. :^]
